### PR TITLE
[DSL] Move auth method to be specified in the project block

### PR DESCRIPTION
**Pull request by c-harding**
Instead of `#auth(method)` in the service block, we now have `#authMethod(method)` in the project block, and bare `#auth` in the service block

### CODE REVIEW

**Review by jaylees14 — APPROVED** [review]
Nice one

File: src/main/scala/temple/ast/Metadata.scala [hunk 1]
```diff
@@ -92,13 +92,16 @@ object Metadata {
...
 
-  object ServiceAuth extends Enum[ServiceAuth] {
-    override def values: IndexedSeq[ServiceAuth] = findValues
-    case object Email extends ServiceAuth("email")
+  object AuthMethod extends Enum[AuthMethod] {
+    override def values: IndexedSeq[AuthMethod] = findValues
+    case object Email extends AuthMethod("email")
   }
 
+  type ServiceAuth = ServiceAuth.type
```
Comment (Lewiky) [inline src/main/scala/temple/ast/Metadata.scala:102]: Why is this line needed? And it's defined before `ServiceAuth` is?

**Review by Lewiky — APPROVED** [review]
Look good

File: src/main/scala/temple/ast/Metadata.scala (hunk 1 above)
Comment (c-harding) [reply src/main/scala/temple/ast/Metadata.scala:102]: So `lookupMetadata` etc work nicely. Order doesn’t really matter
